### PR TITLE
Update netlify to use new API. 

### PR DESCRIPTION
**Pull request by sipsma**
# NOTE!
This also includes a change to support a plaintext field on the secret object. Without this there's no reasonable way for an extension to get a plaintext secret value.

Unreasonable ways include having the extension mount the secret into an exec, cat the value and select the stdout. That defeats the purpose of secrets though.

---

The automated test is skipped because we don't have a shared netlify token; it can still be useful locally by providing your netlify token as an env var if desired.

### CODE REVIEW

**Comment by aluzzardi** [comment]
> This also includes a change to support a plaintext field on the secret object. Without this there's no reasonable way for an extension to get a plaintext secret value.

Yep. I know there's secrets API discussions and TBD's going around, but we'll need something for now to get unstuck

**Review by aluzzardi — APPROVED** [review]
Awesome.

I'll let @vito and @shykes chime in regarding plaintext secrets.

Also, I think you could do me the honor of `git rm sdk/go/dagger/legacy_client.go` -- that was the last remaining bits from the old SDK

**Comment by sipsma** [comment]
> Also, I think you could do me the honor of git rm sdk/go/dagger/legacy_client.go -- that was the last remaining bits from the old SDK

Sounds great :-) I need to update the Go version of netlify too quickly (this is just ts), but I'll push a commit with that rm too.

**Review by aluzzardi — APPROVED** [review]

File: core/integration/secret_test.go [hunk 1]
```diff
@@ -124,3 +127,15 @@ func TestSecretMountFromFileWithOverridingMount(t *testing.T) {
...
+func TestSecretPlaintext(t *testing.T) {
+	ctx := context.Background()
+	c, err := dagger.Connect(ctx)
+	require.NoError(t, err)
+	defer c.Close()
+
+	plaintext, err := c.Core().Directory().
+		WithNewFile("TOP_SECRET", api.DirectoryWithNewFileOpts{Contents: "hi"}).File("TOP_SECRET").Secret().Plaintext(ctx)
+	require.NoError(t, err)
+	require.Equal(t, "hi", plaintext)
```
Comment (vito) [inline core/integration/secret_test.go:140]: Can't wait to use the SDK more in the tests! :drooling_face:

**Review by vito — APPROVED** [review]
Looks great! Adding plaintext makes sense to me. If you already have access to run API queries, there's already some way to expose it (mount + exec like you said), so it doesn't shift the security needle at all.

**Pull request merged**